packages/manager/cypress/e2e/core/account/user-permissions.spec.ts

```diff
@@ -558,4 +558,80 @@ describe('User permission management', () => {
           .click();
       });
   });
+
+  it.only('disables "Ready Only" and "None" and defaults to "Read Write" Billing Access for "Proxy" account users with Parent/Child feature flag', () => {
+    const mockProfile = profileFactory.build({
+      username: 'proxy-user',
+    });
+
+    const mockActiveUser = accountUserFactory.build({
+      username: 'proxy-user',
+      restricted: false,
+      user_type: 'proxy',
+    });
+
+    const mockRestrictedUser = {
+      ...mockActiveUser,
+      restricted: true,
+      username: 'restricted-proxy-user',
+    };
+
+    const mockUserGrants = grantsFactory.build({
+      global: { account_access: 'read_write' },
+    });
+
+    // TODO: Parent/Child - M3-7559 clean up when feature is live in prod and feature flag is removed.
+    mockAppendFeatureFlags({
+      parentChildAccountAccess: makeFeatureFlagData(true),
+    }).as('getFeatureFlags');
+    mockGetFeatureFlagClientstream().as('getClientStream');
+
+    mockGetUsers([mockActiveUser, mockRestrictedUser]).as('getUsers');
+    mockGetUser(mockActiveUser);
+    mockGetUserGrants(mockActiveUser.username, mockUserGrants);
+    mockGetProfile(mockProfile);
+
+    // Navigate to Users & Grants page, find mock restricted user, click its "User Permissions" button.
+    cy.visitWithLogin('/account/users');
+    cy.wait('@getUsers');
+    cy.findByText(mockRestrictedUser.username)
+      .should('be.visible')
+      .closest('tr')
+      .within(() => {
+        ui.button
+          .findByTitle('User Permissions')
+          .should('be.visible')
+          .should('be.enabled')
+          .click();
+      });
+
+    cy.visitWithLogin(
+      `/account/users/${mockRestrictedUser.username}/permissions`
+    );
+    mockGetUser(mockRestrictedUser);
+    mockGetUserGrants(mockRestrictedUser.username, mockUserGrants);
```

Comment:
Minor nitpick since this works, but you can get rid of the call to `cy.visitWithLogin()` and save a little time since `cy.visit` and `cy.visitWithLogin` can be slow.

```suggestion
    cy.url().should('endWith', `/account/users/${mockRestrictedUser.username}/permissions`);
    // Move these calls up with the others around lines 589-592:
    // mockGetUser(mockRestrictedUser);
    // mockGetUserGrants(mockRestrictedUser.username, mockUserGrants);
```
(I think this probably applies to the test above this one, too, but I only messed around with this test)